pilot/pkg/networking/plugin/authz/test_helper.go

```diff
@@ -189,3 +190,72 @@ func generatePolicyWithHTTPMethodAndGroupClaim(methodName, claimName string) *po
 		}},
 	}
 }
+
+// nolint:deadcode
+func generateExpectRBACForSinglePolicy(serviceRoleName string, rbacPolicy *policy.Policy) *policy.RBAC {
+	return &policy.RBAC{
+		Action: policy.RBAC_ALLOW,
+		Policies: map[string]*policy.Policy{
+			serviceRoleName: rbacPolicy,
+		},
+	}
+}
+
+// nolint:deadcode
+func generateSimpleServiceRoleBindingAllGroups(serviceRoleName, serviceRoleBindingName string) model.Config {
+	return model.Config{
+		ConfigMeta: model.ConfigMeta{Name: serviceRoleBindingName},
+		Spec: &rbacproto.ServiceRoleBinding{
+			Subjects: []*rbacproto.Subject{
+				{
+					Properties: map[string]string{
+						"request.auth.claims[groups]": "group*",
+					},
+				},
+			},
+			RoleRef: &rbacproto.RoleRef{
+				Kind: "ServiceRole",
+				Name: serviceRoleName,
+			},
+		},
+	}
+}
+
+// nolint: deadcode
+func generateSimplePolicyForNotRuleWithHeader(header string, exactMatch string) *policy.Policy {
+	return &policy.Policy{
+		Permissions: []*policy.Permission{
+			{
+				Rule: &policy.Permission_AndRules{
+					AndRules: &policy.Permission_Set{
+						Rules: []*policy.Permission{
+							{
+								Rule: &policy.Permission_NotRule{
+									NotRule: &policy.Permission{
+										Rule: generateHeaderRule([]*route.HeaderMatcher{
+											{Name: header, HeaderMatchSpecifier: &route.HeaderMatcher_ExactMatch{ExactMatch: exactMatch}},
+										}),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		Principals: []*policy.Principal{{
+			Identifier: &policy.Principal_AndIds{
+				AndIds: &policy.Principal_Set{
+					Ids: []*policy.Principal{
+						{
+							Identifier: &policy.Principal_Metadata{
+								Metadata: generateMetadataListMatcher(authn.AuthnFilterName,
+									[]string{attrRequestClaims, "groups"}, "group*"),
```

Comment:
NIT: I saw "group*" has been used in the tests in many places. We probably do not need to test the same thing in so many tests. For example, if a test is for "not_methods" and not for testing "principal", let's use the simplest principal (e.g., a single service account).

Reply:
Thanks Limin. Maybe another PR? I still need to submit another small PR to check if the user defines not_ports and destination.port at the same time.

Reply:
Sure. Just a small suggestion. You don't have to make the change now.

Reply:
Thanks!